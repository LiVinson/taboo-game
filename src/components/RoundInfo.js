import React from "react"

export default function RoundInfo({ round, giver, watcher, turn }) {
  const giverTeam = turn === 1 ? "Team 1" : "Team 2"
  const watcherTeam = turn === 2 ? "Team 1" : "Team 2"

  return (
    <div>
      <p>Round: {round}</p>
      <p>
        Giver: {giver} ({giverTeam}) Watcher: {watcher} ({watcherTeam})
      </p>
    </div>
  )
}

// function RoundCard({ text, value }) {}
