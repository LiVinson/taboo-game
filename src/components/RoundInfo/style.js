import styled from "styled-components"

export const StyledRoundInfo = styled.div`
    margin-bottom:1rem;
    width:100%;
    max-width: 40rem;
    display:flex;
    flex-wrap:wrap;
    text-align:center;
    justify-content:space-around;
    margin: 0 auto 1rem auto;
    
`
export const RoundData = styled.p`
    font-size: 2rem;
    font-family: ${props => props.theme.text.display};
    color: ${props => props.theme.color.primary};
    text-transform: uppercase;
    /* Used to set round # to full width while others are dynamic through flex */
    width: ${props => props.width && props.width};
    
    & span {
        color: ${props => props.theme.color.grayDark2};
    }
`
RoundData.displayName = "RoundData"