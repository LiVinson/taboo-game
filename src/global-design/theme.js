const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
