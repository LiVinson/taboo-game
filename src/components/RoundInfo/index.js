import RoundInfo from "./RoundInfo"

export default RoundInfo