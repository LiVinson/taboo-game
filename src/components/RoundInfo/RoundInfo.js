import React from 'react'
import PropTypes from 'prop-types'
import { StyledRoundInfo, RoundNumber, Role } from './style'
import Star from 'components/shared/Star'

const RoundInfo = ({ round, giver, watcher, currentPlayerId }) => {
	return (
		<StyledRoundInfo>
			<RoundNumber>Round: {round}</RoundNumber>
			<div>
				<Role>
					Giver: {giver.name}
					{currentPlayerId === giver.playerId && <Star/>}{' '}
				</Role>
				<Role>
					Watcher: {watcher.name}
					{currentPlayerId === watcher.playerId && <Star/>}
				</Role>
			</div>
		</StyledRoundInfo>
	)
}

RoundInfo.propTypes = {
	round: PropTypes.number.isRequired,
	giver: PropTypes.object.isRequired,
	watcher: PropTypes.object.isRequired,
}

export default RoundInfo
