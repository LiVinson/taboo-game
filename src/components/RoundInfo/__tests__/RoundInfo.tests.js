import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import RoundInfo from '../RoundInfo'

test('RoundInfo renders correctly for active player', () => {
	const props = {
		round: 1,
		giver: { name: 'Sam', playerId: '12345', team: 'team 1' },
		watcher: { name: 'Jo', playerId: '23456', team: 'team 2' },
		currentPlayerId: "12345"
	}
	
	const wrapper = renderer
		.create(
			<ThemeProvider theme={theme}>
				<RoundInfo {...props} />
			</ThemeProvider>
		)
		.toJSON()
	expect(wrapper).toMatchSnapshot()
})

test('RoundInfo renders correctly for non-active player', () => {
	const props = {
		round: 1,
		giver: { name: 'Sam', playerId: '12345', team: 'team 1' },
		watcher: { name: 'Jo', playerId: '23456', team: 'team 2' },
		currentPlayerId: "98765"
	}
	
	const wrapper = renderer
		.create(
			<ThemeProvider theme={theme}>
				<RoundInfo {...props} />
			</ThemeProvider>
		)
		.toJSON()
	expect(wrapper).toMatchSnapshot()
})