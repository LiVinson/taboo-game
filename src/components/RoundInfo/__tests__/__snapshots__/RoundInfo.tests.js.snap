// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundInfo renders correctly for active player 1`] = `
<div
  className="sc-AxjAm jJJSIE"
>
  <p
    className="sc-AxirZ ibSLEF"
  >
    Round: 
    1
  </p>
  <div>
    <p
      className="sc-AxiKw kEKSvH"
    >
      Giver: 
      Sam
      <svg
        className="sc-AxhCb fwzZEW"
      >
        star-full.svg
      </svg>
       
    </p>
    <p
      className="sc-AxiKw kEKSvH"
    >
      Watcher: 
      Jo
    </p>
  </div>
</div>
`;

exports[`RoundInfo renders correctly for non-active player 1`] = `
<div
  className="sc-AxjAm jJJSIE"
>
  <p
    className="sc-AxirZ ibSLEF"
  >
    Round: 
    1
  </p>
  <div>
    <p
      className="sc-AxiKw kEKSvH"
    >
      Giver: 
      Sam
       
    </p>
    <p
      className="sc-AxiKw kEKSvH"
    >
      Watcher: 
      Jo
    </p>
  </div>
</div>
`;
